pkg/predicate/predicate.go

```diff
@@ -268,3 +271,48 @@ func (o or) Generic(e event.GenericEvent) bool {
 	}
 	return false
 }
+
+// LabelSelectorPredicate implements predicate functions and requires a selector that will filter an event by its labels.
+//
+// This predicate will skip reconciliation on all events that have labels that do not match the selector of the predicate.
+// A metav1.LabelSelector is passed as an input parameter while instantiating an LabelSelectorPredicate, and this is (explicitly)
+// converted to a labels.Selector object, and stored within the struct. When a predicate function is evoked, the event labels are checked
+// for a match with the predicate's selector to decide whether an event undergoes reconciliation or not.
+// If the labels match, the event will undergo reconciliation. If the labels do not match, it will skip reconciliation for that particular event.
+
+type LabelSelectorPredicate struct {
+	Funcs
```

Comment:
Why the embedded `Funcs`? Its unused or not? And why is the type exported?

Reply:
I realized that the Funcs are available by default. I will remove them from the struct and un-export the struct too.

Reply:
```
func NewLabelSelectorPredicate(s metav1.LabelSelector)(Predicate, error){
    selector, err := metav1.LabelSelectorAsSelector(&s)
    if err != nil {return nil, err}
    return NewPredicateFuncs(func (o controllerutil.Object) bool {
         selector.Matches(labels.Set(o.GetLabels())
    }
}